src/wallet/crypter.h

```diff
@@ -155,10 +158,19 @@ class CCryptoKeyStore : public CBasicKeyStore
             LOCK(cs_KeyStore);
             result = vMasterKey.empty();
         }
+        // fForMixing   fOnlyMixingAllowed  return
+        // ---------------------------------------
+        // true         true                result
+        // true         false               result
+        // false        true                true
+        // false        false               result
+
+        if(!fForMixing && fOnlyMixingAllowed) return true;
```

Comment:
I think the comment I would add is to the effect that `IsLocked(true)` should be used to determine if the wallet is fully locked so that no operations requiring access to private keys are possible while `IsLocked(false)` should be used to determine if the wallet is available for mixing.

At least I think that's what the logic here is supposed to be.  It took me quite a while to figure that out so I think the comment is very necessary.